### PR TITLE
HTML validation fixes

### PR DESCRIPTION
**Pull request by EreMaijala**
This is a set of changes that allows VuFind's bootstrap3 and sandal themes to produce valid HTML without warnings.

Theme config now includes doctype so that it can be set early before processing layout.phtml. This allows e.g. Laminas Form and HeadScript helper to do the right thing, and makes it possible to leave out the type attribute from script tags.

### CODE REVIEW

File: themes/root/templates/layout/help.phtml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?=$this->doctype('XHTML1_TRANSITIONAL')?>
+<?=$this->doctype()?>
 <html xmlns="http://www.w3.org/1999/xhtml" lang="<?=$this->layout()->userLang ?>" xml:lang="en">
```
Comment (demiankatz) [inline themes/root/templates/layout/help.phtml]: Does it make sense to use the XHTML namespace without an XHTML doctype?

File: module/VuFindTheme/src/VuFindTheme/View/Helper/HeadScript.php [hunk 2]
```diff
@@ -114,6 +114,9 @@ public function itemToString($item, $indent, $escapeStart, $escapeEnd)
         }
 
         $this->addNonce($item);
+        if ($this->view && $this->view->plugin('doctype')->isHtml5()) {
+            $item->type = '';
```
Comment (demiankatz) [inline module/VuFindTheme/src/VuFindTheme/View/Helper/HeadScript.php]: Just out of curiosity, what is the purpose of this change? Perhaps a comment would be worthwhile here.

**Review by demiankatz — COMMENTED** [review]
A couple of questions; I'm also requesting a review from @crhallberg since he may be more familiar with some of this territory than I am.

File: themes/root/templates/layout/help.phtml (hunk 1 above)
Comment (EreMaijala) [reply themes/root/templates/layout/help.phtml]: While it didn't cause any trouble, I don't think we need it. Changed. While at it I noticed that there were a lot of bad anchors in search help pages and replaced those as well.

File: module/VuFindTheme/src/VuFindTheme/View/Helper/HeadScript.php (hunk 2 above)
Comment (EreMaijala) [reply module/VuFindTheme/src/VuFindTheme/View/Helper/HeadScript.php]: It was a misguided attempt at getting rid of the `type="text/javascript"` attribute, which the Laminas helper actually takes care of already. Now removed. Thanks for pointing it out!

**Comment by demiankatz** [comment]
Thanks, @EreMaijala. I've gone ahead and cherry-picked the help translation fixes to dev since they added a lot of noise to this PR and make perfect sense (I also applied them to the new Ukrainian translation that was introduced after you started this PR).

Everything else looks fine to me, but I'm going to leave this open until @crhallberg has a chance to comment, in case he has any other thoughts.

File: themes/bootstrap3/templates/layout/layout.phtml [hunk 3]
```diff
@@ -1,11 +1,12 @@
-<?=$this->doctype('HTML5')?>
+<?=$this->doctype()?>
+
 <html lang="<?=$this->layout()->userLang?>"<?php if ($this->layout()->rtl): ?> dir="rtl"<?php endif; ?>>
   <head>
     <?php $this->setupThemeResources(); ?>
-    <meta charset="utf-8">
```
Comment (demiankatz) [inline themes/bootstrap3/templates/layout/layout.phtml:5]: Oh, one last question: what's the motivation in removing this?

Comment (EreMaijala) [reply themes/bootstrap3/templates/layout/layout.phtml:5]: Validator complained that you shouldn't have it when you already have `<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">`. And that's what headMeta adds.

Comment (demiankatz) [reply themes/bootstrap3/templates/layout/layout.phtml:5]: Okay, that makes perfect sense!

**Review by crhallberg — APPROVED** [review]
All these changes make sense to me!

**Comment by demiankatz** [comment]
Thanks, @crhallberg and @EreMaijala -- just merged this!
